File: iOS/Maryland Daily Trivia/Maryland Daily Trivia/Scoring.swift
//
//  Scoring.swift
//  Maryland Daily Trivia
//
//  Created by Ramon Dominguez on 1/1/26.
//
import Foundation

enum Scoring {

    /// All questions are worth up to 1000 points
    static let maxPoints = 1000

    /// Points based on speed.
    /// - Linearly decreases from 1000 (instant answer) to 0 (timer expires).
    /// - Wrong answer = 0.
    static func points(
        timeLimit: Double,
        secondsRemaining: Double,
        isCorrect: Bool
    ) -> Int {
        guard isCorrect else { return 0 }

        // Clamp ratio to 0...1
        let t = Swift.max(0.0, Swift.min(1.0, secondsRemaining / timeLimit))

        return Int((Double(maxPoints) * t).rounded(.down))
    }
}

File: Android/Android/app/src/main/java/com/copanostudios/marylanddailytrivia/core/Scoring.kt
package com.copanostudios.marylanddailytrivia.core

/**
 * Scoring logic — exact port of iOS Scoring.swift.
 * All questions are worth up to MAX_POINTS.
 * Points decrease linearly from MAX_POINTS (instant) to 0 (timer expires).
 */
object Scoring {
    const val MAX_POINTS = 1000

    /**
     * Calculate points for an answer.
     * @param timeLimit Total time allowed (seconds), e.g. 12.0
     * @param secondsRemaining Time left when answer was given
     * @param isCorrect Whether the answer was correct
     * @return Points earned (0 for wrong answers)
     */
    fun points(timeLimit: Double, secondsRemaining: Double, isCorrect: Boolean): Int {
        if (!isCorrect) return 0
        val t = (secondsRemaining / timeLimit).coerceIn(0.0, 1.0)
        return (MAX_POINTS.toDouble() * t).toInt()
    }
}
